common/src/service/HttpClientHelper.cs

```diff
@@ -40,53 +40,31 @@ internal sealed class HttpClientHelper : IHttpClientHelper
         // IDisposables
 
         private readonly HttpClient _httpClientWithDefaultTimeout;
-
         private readonly HttpClient _httpClientWithNoTimeout;
-        private readonly HttpClientHandler _httpClientHandler;
 
         public HttpClientHelper(
             Uri baseAddress,
             IAuthorizationHeaderProvider authenticationHeaderProvider,
             IDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>> defaultErrorMapping,
             TimeSpan timeout,
-            IWebProxy customHttpProxy)
+            IWebProxy customHttpProxy,
+            int connectionLeaseTimeoutMilliseconds)
```

Comment:
Can we use TimeSpan instead of int?

Reply:
The place where we eventually use this value is as an integer, in milliseconds, so I'd rather not. On top of that, this parameter has special cases when the value is -1 (no connection lease timeout) and 0 (revoke connection lease after every connection)

Reply:
For the first point, it is still recommended to use TimeSpan and in the end to TimeSpan.TotalSeconds. It is less prone to error and easier to read. For the other two, you can use TimeSpan.Zero and TimeSpan.FromSeconds(-1). I recommend to try it out.